modules/db_instance/main.tf

```diff
@@ -1,6 +1,24 @@
 ##############
 # DB instance
 ##############
+
+locals {
+  create_enhanced_role = "${var.monitoring_role_autocreate != "" && var.monitoring_role_arn == ""}"
+}
+
+resource "aws_iam_role" "enhanced_monitoring" {
+  name               = "${var.monitoring_role_autocreate}"
+  assume_role_policy = "${file("${path.module}/policy/enhancedmonitoring.json")}"
+  count              = "${local.create_enhanced_role ? 1 : 0}"
+}
+
+resource "aws_iam_policy_attachment" "enhanced_monitoring" {
+  name       = "${var.monitoring_role_autocreate}-enhanced-monitoring-attachment"
```

Comment:
`name = "${var.monitoring_role_name}"` - shorter is better usually. In this case, it is clear that it is an attachment of enhanced monitoring policy, so `-enhanced-monitoring-attachment` can be safely removed.